iroh/src/node.rs

```diff
@@ -1141,29 +1146,48 @@ impl<D: BaoStore, S: DocStore, C: CollectionParser> RpcHandler<D, S, C> {
         })
     }
 
-    // TODO: streaming
-    async fn bytes_get(self, req: BytesGetRequest) -> RpcResult<BytesGetResponse> {
-        let entry = self
-            .inner
-            .db
-            .get(&req.hash)
-            .ok_or_else(|| RpcError::from(anyhow!("not found")))?;
-        // TODO: size limit
-        // TODO: streaming
-        let data = self.inner.rt.local_pool().spawn_pinned(|| async move {
-            let data = entry
-                .data_reader()
-                .await
-                .map_err(anyhow::Error::from)?
-                .read_to_end()
-                .await
-                .map_err(anyhow::Error::from)?;
-            Result::<_, anyhow::Error>::Ok(data)
+    fn bytes_get(
+        self,
+        req: BytesGetRequest,
+    ) -> impl Stream<Item = RpcResult<BytesGetResponse>> + Send + 'static {
```

Comment:
This works fine and should not be an issue in this case.

But if quic-rpc would optionally give you a sender instead of requiring a signature that returns a stream the additional channel could be avoided.